modules/core/src/main/java/org/apache/ignite/internal/processors/platform/cluster/PlatformClusterNodeFilterImpl.java

```diff
@@ -80,4 +114,14 @@ public void setIgniteInstance(Ignite ignite) {
     public Object getInternalPredicate() {
         return pred;
     }
+
+    @Override
+    public void writeBinary(BinaryWriter writer) throws BinaryObjectException {
```

Comment:
This likely breaks rolling upgrades compatibility, and I'm not sure if we need to override at all.

Reply:
Unfortunately, the whole change is breaking RU and the C++ platform. 
I think this is kind of expected if you are dealing with cache configuration changes?

Reply:
It is fine to say that "if you use this new feature, RU will break". But here we break RU for Service Node Filters, because we change the format.

And this format change looks unnecessary to me, we can remove those overrides.